cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-sync.php

```diff
@@ -172,13 +172,58 @@ public function get_signature( $post_id ) {
 		return $return;
 	}
 
+	/**
+	 * Generate a new Public ID for an asset.
+	 *
+	 * @param int $attachment_id The attachment ID for the new public ID.
+	 *
+	 * @return string|null
+	 */
+	public function generate_public_id( $attachment_id, $prefix = null ) {
+		$settings   = $this->plugin->config['settings'];
+		$cld_folder = trailingslashit( $settings['sync_media']['cloudinary_folder'] );
+		$file       = get_attached_file( $attachment_id );
+		$file_info  = pathinfo( $file );
+		$public_id  = $cld_folder . $prefix . $file_info['filename'];
```

Comment:
I wonder if suffix wouldn't be a better approach to make the public id unique. The reason for that has to do with sorting.
What do you think?

Reply:
@pereirinha On the SEO level, I think that having the name at the end will better benefit the SEO rating.

Reply:
@DavidCramer 

That's interesting. Do you have an article that you can refer to? This way, we document this decision properly, as for us humans, there are other preferred ways to see filenames.

Reply:
@pereirinha - Well, it was the post on seo file names in Cloudinary's docs. It just had descriptive names as the last item.
However, it's minor and the sorting is bigger issue, so let's move it to a suffix. It's cleaner. And even then, we can alter this with the public_id filter anyway.